service_configuration_lib/spark_config.py

```diff
@@ -1125,8 +1127,8 @@ def get_spark_conf(
         spark_conf = {**(spark_opts_from_env or {}), **_filter_user_spark_opts(user_spark_opts)}
         random_postfix = utils.get_random_string(4)
 
-        if aws_creds is not None and aws_creds[2] is not None:
-            spark_conf['spark.hadoop.fs.s3a.aws.credentials.provider'] = AWS_ENV_CREDENTIALS_PROVIDER
+        if (aws_creds is not None and aws_creds[2] is not None) or service_account_name is not None:
+            spark_conf['spark.hadoop.fs.s3a.aws.credentials.provider'] = AWS_DEFAULT_CREDENTIALS_PROVIDER
```

Comment:
`AWS_DEFAULT_CREDENTIALS_PROVIDER` will work with access keys in the env as well